### ISSUE TITLE
Add Namespaces

### ISSUE THREAD

**Issue by HD-CMS** [maclof/kubernetes-client#14]
Hey,

currently I work on a project and need to manage `namespaces`. I would suggest to implement the relevant endpoint/model/repo methods to this library.

What do you guys think?
When okay I would develop the needed stuff and create a merge request.

**Comment by maclof** [comment]
Hey @HD-CMS 

Can you give me an example of what you mean? I'm open to all changes as long as they benefit the project and everybody using it 👍

**Comment by HD-CMS** [comment]
Hey @maclof
I mean to manage namespaces the same way you manage the other resources:

```
use Maclof\Kubernetes\Models\Namespace;

$namespace = new Namespace(['metadata' => ['name' => 'my-namespace']]);
$client->namespaces()->create($namespace);
$client->namespaces()->update($namespace);
$client->namespaces()->delete($namespace);
```

**Comment by maclof** [comment]
Yes please add this, I want to aim for coverage of all Kubernetes resources :)

**Comment by HD-CMS** [comment]
Okay, 

let me do some magic ;-)

**Comment by HD-CMS** [comment]
Sorry @maclof for the late reply, to much other work to do ;-)

I added the namespace resource like the other ones. `Namespace` is reserved for `php` internals, so I used `NamespaceModel` instead.

Also I changed the API-Calls to match always the current `$this->namespace` setting like the existing `$this->find()` method. To match the correct API Schema -> `kind` I added an option to overwrite `$this-schema` in all models.

Feel free to comment and improve ;-)
